src/app/workflow/info-tab/info-tab.service.ts

```diff
@@ -265,13 +265,36 @@ export class InfoTabService {
    * @memberof InfoTabService
    */
   getTRSLink(path: string, versionName: string, descriptorType: string, descriptorPath: string, entryType: EntryType): string {
-    const prefix: string = entryType === EntryType.BioWorkflow ? ga4ghWorkflowIdPrefix : ga4ghServiceIdPrefix;
     return (
-      `${Dockstore.API_URI}${ga4ghPath}/tools/${encodeURIComponent(prefix + path)}` +
+      `${Dockstore.API_URI}${ga4ghPath}/tools/${encodeURIComponent(this.getTRSIDFromPath(path, entryType))}` +
       `/versions/${encodeURIComponent(versionName)}/plain-` +
       descriptorType.toUpperCase() +
       `/descriptor/` +
       descriptorPath
     );
   }
+
+  getTRSId(workflow: Workflow | undefined, entryType: EntryType): string {
+    if (!workflow) {
+      return '';
+    }
+    return this.getTRSIDFromPath(workflow.full_workflow_path, entryType);
+  }
+
+  private getTRSIDFromPath(fullWorkflowPath: string, entryType: EntryType): string {
+    return this.getTRSPrefix(entryType) + fullWorkflowPath;
+  }
+
+  private getTRSPrefix(entryType: EntryType): string {
+    switch (entryType) {
+      case EntryType.BioWorkflow:
+        return ga4ghWorkflowIdPrefix;
+      case EntryType.Service:
+        return ga4ghServiceIdPrefix;
+      case EntryType.Tool: // This one shouldn't get invoked from this code
+      case EntryType.AppTool:
+      default:
+        return '';
```

Comment:
starting to lean toward these cases dying more dramatically to warn us when working on something new, like when we work on notebooks maybe

Reply:
I wonder if someday, we might be able to go a step farther and eliminate the cases somehow...  Maybe enhance the data objects so that they themselves can compute (or are decorated with) values like their TRS urls?